src/uct/ib/mlx5/rc/rc_mlx5_common.c

```diff
@@ -596,21 +601,14 @@ void uct_rc_mlx5_release_desc(uct_recv_desc_t *self, void *desc)
 
 ucs_status_t
 uct_rc_mlx5_dp_ordering_ooo_init(uct_rc_mlx5_iface_common_t *iface,
-                                 uint64_t tl_flag,
+                                 uct_ib_mlx5_dp_ordering_t dp_ordering_cap,
```

Comment:
need to make boolean as a parameter and setting OOO_RW if true in that function

Reply:
I didn't understand why, as dp_ordering_cap would indicate that

Reply:
I see that we now pass the second parameter as a boolean in the PR:
```
    status = uct_rc_mlx5_dp_ordering_ooo_init(&self->super,
                                              (md->dp_ordering_cap.rc >=
                                               UCT_IB_MLX5_DP_ORDERING_OOO_RW),
                                               &config->rc_mlx5_common, "gga");
```

So the the second argument should become boolean, and we could test it and set variables to `UCT_IB_MLX5_DP_ORDERING_OOO_RW` in this function accordingly.

Now `UCT_IB_MLX5_DP_ORDERING_OOO_RW` happens to be 1 and match the boolean value, but it is really a coincidence and we should not rely on.

Reply:
That case is only in gga where ddp is not relevant so we are limiting it here to OOO_RW,
that's why it was `ucs_min(md->dp_ordering_cap.rc, UCT_IB_MLX5_DP_ORDERING_OOO_RW)` to make sure we are not passing OOO_ALL here

Reply:
First it was set to `ucs_min(md->dp_ordering_cap.rc, UCT_IB_MLX5_DP_ORDERING_OOO_RW)`. Since it's only either OOO_ALL or OOO_RW, I suggested to use (... >= OOO_RW). So now it is boolean, and we need to change the prototype of the function.

Reply:
but for `rc_mlx5` and `dc_mlx5` it can be either `IBTA` `OOO_RW` or `OOO_ALL` so I don't see how a boolean cover all of those options

Reply:
I will revert it to `ucs_min` in gga